diff --git a/opensilex-security/src/main/java/org/opensilex/security/authentication/AuthenticationService.java b/opensilex-security/src/main/java/org/opensilex/security/authentication/AuthenticationService.java
--- a/opensilex-security/src/main/java/org/opensilex/security/authentication/AuthenticationService.java	2026-06-11 14:38:31.235587100 +0200
+++ b/opensilex-security/src/main/java/org/opensilex/security/authentication/AuthenticationService.java	2026-06-11 14:38:47.194509400 +0200
@@ -699,7 +699,7 @@
         // Create thread to remove user after expire time
         Thread t = new Thread(() -> {
             try {
-                Thread.sleep(getRenewTokenExpiresInSec());
+                Thread.sleep(getRenewTokenExpiresInSec() * 1000L);
                 LOGGER.debug("Renew token UUID expires: " + uuidURI);
                 schedulerForgotUserPasswordRegistry.remove(uuidURI);
             } catch (InterruptedException ex) {
diff --git a/opensilex-security/src/main/java/org/opensilex/security/authentication/api/AuthenticationAPI.java b/opensilex-security/src/main/java/org/opensilex/security/authentication/api/AuthenticationAPI.java
--- a/opensilex-security/src/main/java/org/opensilex/security/authentication/api/AuthenticationAPI.java	2026-06-11 14:38:31.319498400 +0200
+++ b/opensilex-security/src/main/java/org/opensilex/security/authentication/api/AuthenticationAPI.java	2026-06-11 14:39:36.049667900 +0200
@@ -43,9 +43,6 @@
 import java.io.UnsupportedEncodingException;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Paths;
 import java.util.*;
 
 import static org.opensilex.security.SecurityModule.DEFAULT_SUPER_ADMIN_EMAIL;
@@ -262,11 +259,11 @@
     }
     
     private String getForgotPasswordRedirectUrl(URI userForgottenToken) throws OpenSilexModuleNotFoundException {
-        // get address
-        String redirectUrl = emailService.getOpenSilex().getModuleByClass(ServerModule.class).getBaseURL();
-        
-        redirectUrl = Paths.get(redirectUrl , "app" , EMAIL_RESET_PASSWORD_APP_PATH , URLEncoder.encode(userForgottenToken.toString(), StandardCharsets.UTF_8)).toString();
-        return redirectUrl;
+        String baseURL = emailService.getOpenSilex().getModuleByClass(ServerModule.class).getBaseURL();
+        if (!baseURL.endsWith("/")) {
+            baseURL += "/";
+        }
+        return baseURL + "app/" + EMAIL_RESET_PASSWORD_APP_PATH + "/" + userForgottenToken.toString();
     }
     
      /**
